src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/SHPOValidationService.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.validation
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.Booking
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.CalculableSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.ConsecutiveSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.StandardDeterminateSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service.ImportantDates.SDS_40_COMMENCEMENT_DATE
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service.ImportantDates.SHPO_OFFENCE_COMMENCEMENT_DATE
+
+@Service
+class SHPOValidationService {
+
+  private val shpoOffenceCodes = listOf("SE20005", "SE20006", "SE20012", "SE20012A")
+
+  internal fun validate(booking: Booking): List<ValidationMessage> = booking.getAllExtractableSentences().flatMap {
+    when (it) {
+      is ConsecutiveSentence -> validateConsecutiveSentence(it)
+      is StandardDeterminateSentence -> validateStandardDeterminateSentence(it)
+      else -> emptyList()
+    }
+  }
+
+  private fun validateConsecutiveSentence(consecutiveSentence: ConsecutiveSentence): List<ValidationMessage> {
+    if (consecutiveSentence.sentencedAt <= SDS_40_COMMENCEMENT_DATE &&
+      !isConsecutiveSdsHistoricSentenceInValid(consecutiveSentence)
+    ) {
+      return emptyList()
+    }
+
+    return consecutiveSentence.orderedSentences
+      .filterIsInstance<StandardDeterminateSentence>()
+      .filter { isRelevantOffence(it) && it.offence.committedAt < SHPO_OFFENCE_COMMENCEMENT_DATE }
+      .map {
+        ValidationMessage(
+          ValidationCode.SHPO_INVALID_DATE,
+          listOf(it.caseSequence.toString(), it.lineSequence.toString()),
+        )
+      }
+  }
+
+  private fun validateStandardDeterminateSentence(sentence: StandardDeterminateSentence): List<ValidationMessage> =
+    if (isShpoViolation(sentence)) {
+      listOf(
+        ValidationMessage(
+          ValidationCode.SHPO_INVALID_DATE,
+          listOf(sentence.caseSequence.toString(), sentence.lineSequence.toString()),
+        ),
+      )
+    } else {
+      emptyList()
+    }
+
+  private fun isShpoViolation(sentence: StandardDeterminateSentence): Boolean = isRelevantOffence(sentence) &&
+    (isSdsSentenceInValid(sentence) || isSdsHistoricSentenceInValid(sentence))
+
+  private fun isRelevantOffence(sentence: StandardDeterminateSentence): Boolean =
+    !sentence.isSDSPlus && shpoOffenceCodes.contains(sentence.offence.offenceCode)
+
+  private fun isConsecutiveSdsHistoricSentenceInValid(consecutiveSentence: ConsecutiveSentence): Boolean =
+    consecutiveSentence.sentencedAt.isBefore(SDS_40_COMMENCEMENT_DATE) &&
+      consecutiveSentence.sentenceCalculation.releaseDate.isAfter(SDS_40_COMMENCEMENT_DATE)
+
+  private fun isSdsSentenceInValid(sentence: CalculableSentence): Boolean =
```

Comment:
My apologies if I caused this, but it was how the ticket was written....

So this function and the next can be combined into one and the detection of SDS40 commenecement removed - in both scenarios we are doing the same check, which is, is the SHPO offence dated before the SHPO commencement date.

Reply:
After discussion with Scott this was intentional, in order to limit the scope of those affected by the change.
We do have other validation for SA 20 offences being incorrectly dated, but didn't want to have to do the impact analysis of number of sentences that it would impact now.